# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBCursorType.cpp
# Commit: e65d7e9dea10
# Full Hash: e65d7e9dea10c417588224abb452aa2d181e8e0c
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-01-13 15:45:24
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Convert IDBCursor and BackgroundCursorChild to templates to increase type safety and reduce state. r=dom-workers-and-storage-reviewers,ytausky
#   
#   This also simplifies delegating calls that are dependent on the cursor type.
#   
#   Also reduce dependency on IDBCursor.h by moving enums and type traits to IDBCursorType.h
# ==============================================================================

diff -r 422de232c5ce -r e65d7e9dea10 dom/indexedDB/IDBCursorType.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/indexedDB/IDBCursorType.cpp	Fri Jan 10 15:23:52 2020 +0000
@@ -0,0 +1,35 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "IDBCursorType.h"
+
+namespace mozilla {
+namespace dom {
+CommonCursorDataBase::CommonCursorDataBase(Key aKey) : mKey{std::move(aKey)} {}
+
+IndexCursorDataBase::IndexCursorDataBase(Key aKey, Key aLocaleAwareKey,
+                                         Key aObjectStoreKey)
+    : CommonCursorDataBase{std::move(aKey)},
+      mLocaleAwareKey{std::move(aLocaleAwareKey)},
+      mObjectStoreKey{std::move(aObjectStoreKey)} {}
+
+ValueCursorDataBase::ValueCursorDataBase(StructuredCloneReadInfo&& aCloneInfo)
+    : mCloneInfo{std::move(aCloneInfo)} {}
+
+CursorData<IDBCursorType::ObjectStore>::CursorData(
+    Key aKey, StructuredCloneReadInfo&& aCloneInfo)
+    : ObjectStoreCursorDataBase{std::move(aKey)},
+      ValueCursorDataBase{std::move(aCloneInfo)} {}
+
+CursorData<IDBCursorType::Index>::CursorData(
+    Key aKey, Key aLocaleAwareKey, Key aObjectStoreKey,
+    StructuredCloneReadInfo&& aCloneInfo)
+    : IndexCursorDataBase{std::move(aKey), std::move(aLocaleAwareKey),
+                          std::move(aObjectStoreKey)},
+      ValueCursorDataBase{std::move(aCloneInfo)} {}
+
+}  // namespace dom
+}  // namespace mozilla